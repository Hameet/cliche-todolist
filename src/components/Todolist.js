import React from 'react'
import { addIndex, map } from 'ramda'

const indexedMap = addIndex(map)

export default function Todolist ({todos = []}) {
    console.log("hey", indexedMap)
    return (
        <ul>
            { indexedMap(
                ({todo}, idx) => (
                    <li key={idx} data-index={idx}>
                        {todo}
                    </li>
                    ),
                    todos
                )
            }
        </ul>
    )
}