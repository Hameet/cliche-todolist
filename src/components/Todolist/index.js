import React from 'react'
import { addIndex, map } from 'ramda'
import { isNilOrEmpty } from 'ramda-adjunct'

const indexedMap = addIndex(map)

export default function Todolist ({ todos }) {
    // console.log("hey", indexedMap)
    return isNilOrEmpty(todos)
        ? null
        : <ul>
            { indexedMap(
                ({todo}, idx) => (
                    <li key={idx} data-index={idx}>
                        {todo}
                    </li>
                    ),
                    todos
                )
            }
        </ul>
}