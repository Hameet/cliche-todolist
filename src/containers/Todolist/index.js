import { connect } from 'react-redux'

import Todolist from '../../components/Todolist'
import { getTodos, deleteTodo } from '../../state'

function mapStateToProps (state) {
  return {
    todos: getTodos(state)
  }
}

function mapDispatchToProps (dispatch, props) {
  return {
    handleOnDelete: index => {
      dispatch(deleteTodo(index))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Todolist)
