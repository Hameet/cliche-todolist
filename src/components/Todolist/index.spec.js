import React from 'react'
import { shallow } from 'enzyme'

import Todolist from '.'

describe('components:Todolist', function () {
  it('returns null when the Todolist is empty or undefined', () =>
    expect(toJson(shallow(<Todolist/>))
  ).toMatchSnapshot())

  it('renders the Todolist properly', () => {
    const todoslist = [{todo: 'X'}, {todo: 'Y'}]
  
    expect(toJson(shallow(<Todolist todos={todoslist} />))
  ).toMatchSnapshot()})
})

