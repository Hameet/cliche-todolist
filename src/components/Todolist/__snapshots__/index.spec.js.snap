// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:Todolist renders the Todolist properly 1`] = `
<ul>
  <li
    data-index={0}
    key="0"
  >
    X
  </li>
  <li
    data-index={1}
    key="1"
  >
    Y
  </li>
</ul>
`;

exports[`components:Todolist returns null when the Todolist is empty or undefined 1`] = `""`;
